data/src/main/java/me/fontys/semester4/data/entity/Category.java

```diff
@@ -1,36 +1,47 @@
 package me.fontys.semester4.data.entity;
 
-import javax.persistence.Column;
-import javax.persistence.Entity;
-import javax.persistence.Id;
-import javax.persistence.Table;
-
+import javax.persistence.*;
+import java.io.Serial;
 import java.io.Serializable;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 @Entity
 @Table(name = "category")
 public class Category implements Serializable {
 
+    @Serial
     private static final long serialVersionUID = 1L;
 
     @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     @Column(name = "catid", nullable = false)
     private Long catid;
 
-    @Column(name = "parentid")
-    private Long parentid;
-
     @Column(name = "name")
     private String name;
 
-    protected Category() {
+    @ManyToOne(cascade = {CascadeType.ALL})
+    @JoinColumn(referencedColumnName="catid")
+    private Category parent;
 
-    }
+    @OneToMany(mappedBy = "parent", cascade = {CascadeType.ALL})
+    private List<Category> children;
+
+    @ManyToMany(mappedBy = "categories", cascade = {CascadeType.ALL})
+    private Set<Product> products;
 
-    public Category(Long catid, Long parentid, String name) {
+    public Category(Long catid, Category parent, String name) {
         this.catid = catid;
-        this.parentid = parentid;
+        this.parent = parent;
         this.name = name;
+        products = new HashSet<>();
+    }
+
+    public Category() {
+
+        products = new HashSet<>();
```

Comment:
This could call the public constructor and pass null values, to re-use the other constructor and not duplicate construction code

Reply:
Protected constructor now calls public constructor. Commit ready to be pushed.